src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -68,28 +77,44 @@ internal async Task InitializeRpcServerAsync()
 
         internal Task InitializeChannelsAsync()
         {
-            string workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
-            if (_environment.IsLinuxAppServiceEnvironment())
+            if (IsLinuxEnvironment())
             {
```

Comment:
The check here doesn't actually determines the OS as it is currently implemented, it instead determines whether or not you're running on an App Serivice/Azure hosted Linux environment. It would be good to use names that accurately represent that. (e.g. in any Linux scenario outside of App Service/Azure, you'd end up running the `WindowsOperatingSystem` branch). This treats everything else as Windows, which is not accurate.

Reply:
Good point. I missed this earlier. White listing language here indicates language worker needs to be started at application level.  Remove platform check when not in  InPlaceholderMode.

Reply:
@pragnagopa, could you elaborate more on what you mean by "Remove platform check when not in InPlaceholderMode". Even if the worker runtime is set, aren't we white listing languages separately based on what platform we are running on?

Reply:
White listing is only when hosted in Azure specifically to start java worker when running in Antares only and start python worker when running in Azure Linux platforms. Locally all languages are supported across all platforms

Reply:
@fabiocav, discussed with @pragnagopa offline and came to the conclusion that we should have three different classifications for starting language workers at the application level: (1) Linux hosting, (2) Windows hosting and (3) generally white listed applications which will start at the application level. I have made code changes to reflect these three different levels of classification. Category (3) currently only contains Java because the local development experience for Python (regardless of OS platform) relies on file changes being picked up with a job host restart. Starting Python at the app level locally would be inconsistent with the local development experience that we have today.